addons/portal/controllers/portal.py

```diff
@@ -202,7 +202,12 @@ def account(self, redirect=None, **post):
                         values[field] = False
                 values.update({'zip': values.pop('zipcode', '')})
                 self.on_account_update(values, partner)
+                login = values["login"]
+                del values["login"]
                 partner.sudo().write(values)
+                if request.env.user.login != login:
+                    request.env.user.write({"login": login})
+                    request.session.logout()
```

Comment:
Not sure, but here we can sync with the change_password method by reloading the client instead logout.
![image](https://github.com/user-attachments/assets/be2c44b4-c294-4666-a307-0707b5a79eca)

Reply:
@divy-odoo Thanks! we definitely need something to avoid that error or session does not expire first of all(best case scenario).